### PR TITLE
Re-adds revs to the rotation.

### PR DESCRIPTION
**Pull request by ShiftyRail**
Should just be ready for testing.
We have the ~~luxury~~ misfortune of having several event handler systems and I choose the one which seemed the most fit to the task.

Needs to be live-tested of course.
Uses the same threat levels requirement as Cult right now.

:cl:
- rscadd: Re-adds revolution to the gamemode rotation, with the same threat requirements as cult.

### CODE REVIEW

File: code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm [hunk 1]
```diff
@@ -276,3 +276,38 @@
...
+//               REVS		                ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/roundstart/revs
+	name = "Revolution"
+	role_category = ROLE_REV
+	protected_from_jobs = list("Merchant")
+	restricted_from_jobs = list("AI", "Cyborg", "Mobile MMI", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Chaplain", "Head of Personnel", "Internal Affairs Agent", "Chaplain")
+	enemy_jobs = list("AI", "Cyborg", "Security Officer","Detective","Head of Security", "Captain", "Chaplain")
```
[the remark this answers is not in the document]
Comment (jknpj) [reply code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm]: copypaste from cult most likely

[the remark this answers is not in the document]
Comment (ShiftyRail) [reply code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm]: Was fixed

File: code/datums/gamemode/factions/syndicate/rev.dm [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		var/datum/objective/target/assassinate/A = new(auto_target = FALSE)
+		if(A.set_target(head_mind))
+			AppendObjective(A)
+
+#define ALL_HEADS_DEAD 1
+#define ALL_REVS_DEAD 2
+#define SHUTTLE_LEFT 3
+
+/datum/faction/revolution/check_win()
+	// -- 1. Did the shuttle left ?
```
Comment (DamianX) [inline code/datums/gamemode/factions/syndicate/rev.dm]: `left` -> `leave`

File: code/datums/gamemode/factions/faction.dm [hunk 3]
```diff
@@ -84,7 +84,7 @@ var/list/factions_with_hud_icons = list()
 	if(M.GetRole(late_role))
 		WARNING("Mind already had a role of [late_role]!")
 		return 0
-	var/datum/role/R = new roletype(null,src,initial_role) // Add him to our roles
+	var/datum/role/R = new roletype(null,src,late_role) // Add him to our roles
```
Comment (ShiftyRail) [inline code/datums/gamemode/factions/faction.dm:87]: Worthy of being highlighted

File: code/datums/gamemode/objectives/target/target.dm [hunk 4]
```diff
@@ -5,6 +5,11 @@
 	var/auto_target = TRUE //Whether we pick a target automatically on PostAppend()
 	name = ""
 
+/datum/objective/target/New(var/text,var/auto_target = TRUE, var/mob/user = null)
+	src.auto_target = auto_target
```
Comment (ShiftyRail) [inline code/datums/gamemode/objectives/target/target.dm:9]: Worthy of being highlighted too

**Comment by ShiftyRail** [comment]
I ended up doing most tests on local, and it works.

**Comment by DeityLink** [comment]
More roles and variety is good. I really don't like rev personally however. An idea I had that apparently has already been implemented on other servers is having two competing groups of revolutionary, instead of one.

File: code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm [hunk 5]
```diff
@@ -276,3 +276,53 @@
...
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/roundstart/revs
+	name = "Revolution"
+	role_category = ROLE_REV
+	protected_from_jobs = list("Merchant")
+	restricted_from_jobs = list("AI", "Cyborg", "Mobile MMI", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Chief Engineer", "Chief Medical Officer", "Research Director", "Internal Affairs Agent")
+	enemy_jobs = list("AI", "Cyborg", "Security Officer","Detective","Head of Security", "Captain", "Warden")
+	required_enemies = list(2,2,1,1,1,1,1,0,0,0)
+	required_candidates = 1
```
Comment (ShiftyRail) [inline code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm]: I need to fix this, this is a placeholder value for testing.

**Pull request merged**